packages/roomkit-react/src/Prebuilt/components/AuthToken.tsx

```diff
@@ -20,33 +24,48 @@ import { APP_DATA } from '../common/constants';
  * auth_token=123 => uses the passed in token to join instead of fetching from token endpoint
  * ui_mode=activespeaker => lands in active speaker mode after joining the room
  */
-const AuthToken = React.memo(({ authTokenByRoomCodeEndpoint, defaultAuthToken }) => {
+const AuthToken = React.memo<{
+  authTokenByRoomCodeEndpoint: string;
+  defaultAuthToken?: string;
+  activeState?: PrebuiltStates;
+}>(({ authTokenByRoomCodeEndpoint, defaultAuthToken, activeState }) => {
   const hmsActions = useHMSActions();
   const { roomCode, userId } = useHMSPrebuiltContext();
   const [error, setError] = useState({ title: '', body: '' });
-  let authToken = defaultAuthToken;
-  const [, setAuthTokenInAppData] = useSetAppDataByKey(APP_DATA.authToken);
-  const [savedUserId, setSavedUserId] = useSessionStorage(UserPreferencesKeys.USER_ID);
+  const authToken = defaultAuthToken;
+  const [tokenInAppData, setAuthTokenInAppData] = useSetAppDataByKey(APP_DATA.authToken);
+  const [savedUserId, setSavedUserId] = useSessionStorage<string>(UserPreferencesKeys.USER_ID);
+  const progressRef = useRef<boolean | null>(null);
 
   useEffect(() => {
     if (authToken) {
       setAuthTokenInAppData(authToken);
       return;
     }
 
-    if (!roomCode) {
+    if (tokenInAppData || progressRef.current || activeState === PrebuiltStates.LEAVE) {
       return;
     }
 
-    if (!savedUserId && !userId) {
-      setSavedUserId(uuid());
+    if (!roomCode) {
+      console.error('room code not provided');
       return;
     }
 
+    let userIdForAuthToken = userId || savedUserId;
```

Comment:
Can be simplified to this?
```suggestion
    let userIdForAuthToken = userId || savedUserId || uuid();
```

Reply:
it has to be created only once and to be store in storage when we don't have userId